src/crypto/tls/handshake_client_tls13.go

```diff
@@ -480,8 +521,14 @@ func (hs *clientHandshakeStateTLS13) readServerCertificate() error {
 		c.sendAlert(alertIllegalParameter)
 		return errors.New("tls: certificate used with invalid signature algorithm")
 	}
+
+	pk := c.peerCertificates[0].PublicKey
```

Comment:
A few lines before this the signature algorithm is checked for being supported by the client. If I read the DC spec correctly, the client _separately_ advertises supported delegated credential signature algorithms. AFAICT these are currently not verified.

Reply:
This should probably also be checked when verifying the HRR if the `supportedSignatureAlgorithms` changed — basically everywhere that `supportedSignatureAlgorithms` or `supportedSignatureAlgorithmsCert` occur there should possibly be a DC variant.

Reply:
That is true indeed.

Reply:
Should be fixed now.

Reply:
Could you point me to where `isSupportedSignatureAlgorithm` is called for DCs?

Reply:
Oh, right. Should be added now. I thought this was referring to the inside check, but I know what you meant now ;)